### PR TITLE
add APIs and enums for cl_intel_unified_shared_memory

### PR DESCRIPTION
**Pull request by bashbaug**
Adds APIs and enums for the `cl_intel_unified_shared_memory` extension.  The draft extension spec can be found [here](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/USM/cl_intel_unified_shared_memory.asciidoc).

Also, fixes a few other minor issues:

1. Updated copyright dates to 2020.
2. Removed extraneous semi-colons (supersedes #53).
3. Changed a few C++ comments (`//`) to C comments (`/* */`).

### CODE REVIEW

**Comment by bashbaug** [comment]
Thanks!

I just created a PR to update the XML file here: https://github.com/KhronosGroup/OpenCL-Docs/pull/192

I've also fixed the headers so CI builds are passing when compiling pre-OpenCL 1.2 (thanks for setting this up!).

**Comment by Kerilk** [comment]
Hello,

In this PR you wrote the `void` keyword as `<type>void</type>`. Until now, in the xml file, the type markup was not used around `void`.

**Comment by bashbaug** [comment]
Ping... is anything more needed for this PR?  The XML changes have already been merged.

File: CL/cl_ext_intel.h [hunk 1]
```diff
@@ -416,6 +416,275 @@ typedef cl_uint cl_diagnostics_verbose_level;
...
+typedef cl_uint cl_mem_advice_intel;
+
+/* cl_mem_advice_intel */
+/* Enum values 0x4208-0x420F are reserved for future memory advices. */
+
+typedef cl_bitfield cl_mem_migration_flags_intel;
+
+/* cl_mem_migration_flags_intel - bitfield */
+#define CL_MIGRATE_MEM_OBJECT_HOST_INTEL              (1 << 0)
+#define CL_MIGRATE_MEM_OBJECT_CONTENT_UNDEFINED_INTEL (1 << 1)
```
Comment (kpet) [inline CL/cl_ext_intel.h]: These don't seem to be defined in the XML.

**Comment by bashbaug** [comment]
I'm looking into options that do not require defining aliases for the `cl_mem_migration_flags` type and enums, so I've marked this PR WIP for now.

Related spec issue: http://github.com/KhronosGroup/OpenCL-Docs/issues/214

**Comment by bashbaug** [comment]
To get this PR un-stuck for now, let's back out the `cl_mem_migration_flags_intel` aliases and instead require OpenCL 1.2 headers for `clEnqueueMigrateMemINTEL` so we can reuse the existing `cl_mem_migration_flags` type and enums.  This covers all of our current USM devices, and we can always add aliases later for pre-1.2 devices if needed, pending the resolution of https://github.com/KhronosGroup/OpenCL-Docs/issues/214.

I will create a PR to back out `cl_mem_migration_flags_intel` from the XML file, so it remains consistent with the headers.

I've removed WIP and I believe this is ready for review and merging.  Thanks!

File: CL/cl_ext_intel.h (hunk 1 above)
Comment (bashbaug) [reply CL/cl_ext_intel.h]: I've backed these aliases out completely - thanks!

**Comment by bashbaug** [comment]
Thanks!  Merging.

**Pull request merged**